langchain4j/src/main/java/dev/langchain4j/store/embedding/WeaviateEmbeddingStore.java

```diff
@@ -0,0 +1,98 @@
+package dev.langchain4j.store.embedding;
+
+import dev.langchain4j.data.embedding.Embedding;
+import dev.langchain4j.data.segment.TextSegment;
+import java.lang.reflect.Constructor;
+import java.lang.reflect.InvocationTargetException;
+import java.util.List;
+import lombok.Builder;
+
+public class WeaviateEmbeddingStore implements EmbeddingStore<TextSegment> {
+
+  private final EmbeddingStore<TextSegment> implementation;
+
+  @Builder
+  public WeaviateEmbeddingStore(String apiKey, String scheme, String host, String objectClass) {
```

Comment:
Could you pelase add javadoc for each parameter and where to find it, since it is very confusing for someone looking at it for the first time. I will do the same for pinecone.

Reply:
done